gum/arch-arm64/gumarm64writer.c

```diff
@@ -982,3 +996,94 @@ gum_arm64_writer_describe_reg (GumArm64Writer * self,
   }
   ri->index = ri->meta - GUM_MREG_R0;
 }
+
+void
+gum_arm64_writer_put_push_all_registers(GumArm64Writer * self)
+{
+  gum_arm64_writer_put_push_reg_reg(self, ARM64_REG_X0, ARM64_REG_X1);
+  gum_arm64_writer_put_push_reg_reg(self, ARM64_REG_X2, ARM64_REG_X3);
+  gum_arm64_writer_put_push_reg_reg(self, ARM64_REG_X4, ARM64_REG_X5);
+  gum_arm64_writer_put_push_reg_reg(self, ARM64_REG_X6, ARM64_REG_X7);
+  gum_arm64_writer_put_push_reg_reg(self, ARM64_REG_X8, ARM64_REG_X9);
+  gum_arm64_writer_put_push_reg_reg(self, ARM64_REG_X10, ARM64_REG_X11);
+  gum_arm64_writer_put_push_reg_reg(self, ARM64_REG_X12, ARM64_REG_X13);
+  gum_arm64_writer_put_push_reg_reg(self, ARM64_REG_X14, ARM64_REG_X15);
+  gum_arm64_writer_put_push_reg_reg(self, ARM64_REG_X16, ARM64_REG_X17);
+  gum_arm64_writer_put_push_reg_reg(self, ARM64_REG_X18, ARM64_REG_X19);
+  gum_arm64_writer_put_push_reg_reg(self, ARM64_REG_X20, ARM64_REG_X21);
+  gum_arm64_writer_put_push_reg_reg(self, ARM64_REG_X22, ARM64_REG_X23);
+  gum_arm64_writer_put_push_reg_reg(self, ARM64_REG_X24, ARM64_REG_X25);
+  gum_arm64_writer_put_push_reg_reg(self, ARM64_REG_X26, ARM64_REG_X27);
+  gum_arm64_writer_put_push_reg_reg(self, ARM64_REG_X28, ARM64_REG_X29);
+  gum_arm64_writer_put_instruction(self, 0xD53B420F);//MRS X15, NZCV
```

Comment:
C style comments only. And always a space between code and comment, and between comment start and first character of comment.